pulp_container/app/registry_api.py

```diff
@@ -280,6 +284,44 @@ def get_drv_pull(self, path):
             raise RepositoryNotFound(name=path)
         return distribution, distribution.repository, repository_version
 
+    def get_pull_through_drv(self, path):
+        root_cache_distribution = (
+            models.ContainerPullThroughDistribution.objects.annotate(path=Value(path))
+            .filter(path__startswith=F("base_path"))
+            .order_by("-base_path")
+            .first()
+        )
+        if not root_cache_distribution:
+            raise RepositoryNotFound(name=path)
+
+        try:
+            with transaction.atomic():
+                cache_repository, _ = models.ContainerRepository.objects.get_or_create(
+                    name=path, retain_repo_versions=1
+                )
+
+                upstream_name = path.split(root_cache_distribution.base_path, maxsplit=1)[1]
+                cache_remote, _ = models.ContainerRemote.objects.get_or_create(
```

Comment:
I need help here to understand. With this change you made a regular sync repo pull-through cache as well. I don't remember if this was intended or not.

I thought you will have pull-through-remote associate to special distribution only, and that remote will be used during pull-through requests.
Now i see the in-memory-download on the regular remote and not on special one?

Reply:
Yes, a regular remote is used for the pull-through caching. The special remote comes from the idea that we want a dedicated remote managed inside pulp-container, but with no reference to `upstream-name`. This remote is meant to be attached to a special pull-through distribution that helps us build regular remotes on the fly when processing incoming requests for pull-throughs.